ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/JdbcLedgerDao.scala

```diff
@@ -655,22 +655,54 @@ private class JdbcLedgerDao(
     logger.info("Storing package entry")
     dbDispatcher.executeSql(metrics.daml.index.db.storePackageEntryDbMetrics) {
       implicit connection =>
-        val update = optEntry.map {
-          case PackageLedgerEntry.PackageUploadAccepted(submissionId, recordTime) =>
-            Update.PublicPackageUpload(
-              archives = packages.view.map(_._1).toList,
-              sourceDescription = packages.headOption.flatMap(
-                _._2.sourceDescription
-              ), // this is valid since all clients of this method share the same behavior: all PackageDetails-s will have the same sourceDescription
-              recordTime = Time.Timestamp.assertFromInstant(recordTime),
-              submissionId = Some(submissionId),
+        // Note on knownSince and recordTime:
+        // - There are two different time values in the input: PackageDetails.knownSince and PackageUploadAccepted.recordTime
+        // - There is only one time value in the intermediate values: PublicPackageUpload.recordTime
+        // - There are two different time values in the database schema: packages.known_since and package_entries.recorded_at
+        // This is not an issue since all callers of this method always use the same value for all knownSince and recordTime times.
+        //
+        // Note on sourceDescription:
+        // - In the input, each package can have its own source description (see PackageDetails.sourceDescription)
+        // - In the intermediate value, there is only one source description for all packages (see PublicPackageUpload.sourceDescription)
+        // - In the database schema, each package can have its own source description (see packages.source_description)
+        // This is again not an issue since all callers of this method always use the same value for all source descriptions.
+        val update = optEntry match {
+          case None =>
```

Comment:
we seem to have no test case for this in conformance tests. do you think we have in the Jdbc test suits? shall we have a TODO to add one?

Reply:
We do cover this in [JdbcLedgerDaoPackagesSpec](https://github.com/digital-asset/daml/blob/main/ledger/participant-integration-api/src/test/lib/scala/platform/store/dao/JdbcLedgerDaoPackagesSpec.scala#L104). These tests are not run with the append-only schema though.

I'm not 100% sure, but I don't think it's possible to cover this in the conformance test.